### PR TITLE
upload spin linux amd64 artifacts to S3

### PR DESCRIPTION
**Pull request by rajatjindal**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -178,6 +178,19 @@ jobs:
           file: _dist/spin-${{ env.RELEASE_VERSION }}-${{ env.RUNNER_OS }}-${{ matrix.config.arch }}.zip
           tag: ${{ github.ref }}
 
+      - name: Configure AWS Credentials
+        if: runner.os == 'linux' && matrix.config.arch == 'amd64'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/${{ secrets.INFRA_NAMESPACE }}-${{ secrets.AWS_REGION }}-spin-release-artifacts
```
Comment (endocrimes) [inline .github/workflows/release.yml:185]:
```suggestion
          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/spin-release-artifacts-${{ secrets.AWS_REGION }}
```

Comment (rajatjindal) [reply .github/workflows/release.yml]: ✅, dropped region also from the role name

**Comment by itowlson** [comment]
@rajatjindal could you say more about the motivation for this?

**Comment by rajatjindal** [comment]
Hi Ivan, currently we download spin artifacts for internal testing purposes and those skews the GitHub statistics for the releases. 

so we wanted to have a different source for downloading these artifacts for internal purposes.

**Comment by rajatjindal** [comment]
Hi @itowlson, do you see any concerns with this or we can merge this?

**Comment by itowlson** [comment]
No, sorry, I just wanted to understand why, and now I do.  Sorry for not closing the loop.

**Pull request merged**